### DISCUSSION TITLE
Autocomplete with GraphQL

### DISCUSSION THREAD

**Discussion by mahajansaket** [prisma/prisma#5900] [Q&A]
https://stackoverflow.com/q/66416549/6519391

I'm using React Typescript and have a Autocomplete Material UI component. I'm trying to get query suggestions into Autocomplete component.

My graphql queries looks like this:

Query Definition:

```
import gql from 'graphql-tag';
import {useQuery} from 'react-apollo';
import {Autocomplete} from '@material-ui/lab';
const TODOS = gql`
  query todos($id: ID!) {
    todo(id: $id) {
      id
      name
    }
  }
`;
```

Query Initialization:

```
const { loading, error, data } = useQuery(TODOS, {
  variables: { id: 1 }
});
```

Autocomplete component

```
{loading ? (
                <p>Loading ...</p>
              ) : (
                <>
                  <div>
                    {data &&
                  data.todo &&
                  data.todo.map(
                    (idx: {id: number; name: string}) => (
                      <Autocomplete
                        id="combo-box-demo"
                        options={idx.name}
                        getOptionLabel={option => option.name}
                        style={{width: 300}}
                        renderInput={params => (
                          <TextField
                            {...params}
                            label="Combo box"
                            variant="outlined"
                          />
                        )}
                      />
                    )
                  )}
```

I'm receiving the following errors:

    For line: "`options={idx.name}`" - Type 'string' is not assignable to type 'unknown[]'.
    For line: "`getOptionLabel={option => option.name}`" - Object is of type 'unknown'.

I had the following problems:

    How can I get this working?
    What is causing this error?

**Comment by janpio** [comment]
How is this question related to Prisma? I am not seeing how we could help here.

**Reply by pantharshit00** [reply]
Indeed, I will suggest @mahajansaket to open this issue in the apollo repository

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
